src/packages/core/src/server.ts

```diff
@@ -125,10 +134,16 @@ export class Server<
     //   const server = Ganache.server();
     //   const provider = server.provider;
     //   await server.listen(8545)
-    const connector = (this.#connector = ConnectorLoader.initialize(
-      this.#providerOptions
-    ));
-    this.#initializer = this.initialize(connector);
+    const loader = ConnectorLoader.initialize(this.#providerOptions);
+    const connector = (this.#connector = loader.connector);
+
+    // Since the ConnectorLoader starts an async promise that we intentionally
+    // don't await yet we keep the promise around for something else to handle
+    // later.
+    this.#initializer = Promise.all([
```

Comment:
this promise is awaited in the `listen` function below, and that is where a Connector/Provider's initialization rejections will get handled now.